pir/pircl/shard_cl.go

```diff
@@ -24,6 +26,54 @@ type ShardCL struct {
 	clData     cl.Mem
 }
 
+// NewShard creates a new cuda shard conforming to the common interface
+func NewShard(bucketSize int, data []byte, userdata string) pir.Shard {
```

Comment:
it's a string passed on the command line. since it's implementation specific seemed easier to let the implementations figure out how to handle it.